core/src/test/java/nl/nn/adapterframework/util/SizeLimitedVectorTest.java

```diff
@@ -0,0 +1,55 @@
+package nl.nn.adapterframework.util;
+
+import org.junit.Test;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.fail;
+
+/**
+ * SizeLimitedVector Tester.
+ *
+ * @author <Sina Sen>
+
+ */
+public class SizeLimitedVectorTest {
+
+
+    /**
+     *
+     * Method: add(Object o)
+     *
+     */
+    @Test
+    public void testAdd() throws ArrayIndexOutOfBoundsException {
+        SizeLimitedVector slv = new SizeLimitedVector(10);
+        slv.add("testString");
+        assertEquals(slv.getMaxSize(), 10);
+        assertEquals(slv.get(0), "testString");
+
+    }
+
+
+    @Test
+    public void testMaxSizePassed() throws ArrayIndexOutOfBoundsException {
+        try{SizeLimitedVector slv = new SizeLimitedVector(1);
+        slv.add(13);
+        slv.add(14);
+        }
+        catch (ArrayIndexOutOfBoundsException e) {
+            fail("array capacity is passed, capacity was 1, but 2 elements got added");
+        }
```

Comment:
What exactly are you testing here? this catch clause should never happen?
Maybe add some test assertions :)

Reply:
why shouldn't the catch clause happen? The second add operation passes the size of the vector thus raises and AIOoBounds exception. I'm testing the case that if one adds more items to the vector than the size obliges, what happens next. In the original method .add there was not such a case that raises an expception so I used my own one.

Reply:
The whole point of the SizeLimitedVector is that you can append as many things to it as you like but it only stores the last X amount.

Reply:
yes, I changed the test accordingly.